site/app/templates/course/CourseMaterials.twig

```diff
@@ -64,6 +64,11 @@
             </a>
             {% if userGroup == 1 %}
                 <a onclick='newDeleteCourseMaterialForm("{{ core.buildUrl({'component': 'misc', 'page': 'delete_course_material_folder', 'dir': 'course_materials', 'file': dir, 'path': folderPath }) }}", "{{ dir }}");'> <i class="fas fa-trash" aria-hidden="true" style="font-size: 16px; margin: 5px;"></i></a>
+                <input id="share_all_checkbox_{{ id }}" class="{{ parentId }}" type="checkbox"
+                        {% if ( (fileShares[path] == "1") )%}
+                            checked="true"
+                        {% endif %}
+                       onclick="shareAllToOther('{{ id }}', '{{ path }}')"/><label for="latest-events">Share All to student on</label>
```

Comment:
If you want have the label "Share All to student on", it implies that the user should be able to specify a date to share the whole folder. 

See code for sharing a single file
```suggestion
onclick="shareToOther('{{ id }}', '{{ path }}')" name="ch1"/><label for="latest-events">Share to student on</label>
                <input name="release_date" id="date_to_release_{{ id }}" class="date_picker" type="text" size="26" value="{{ fileReleaseDates[path]}}" onChange="setNewDateTime(this, '{{ path }}')"/>
```
Notice the above input field. 
Additionally, if you choose to share a whole folder, the release dates of the children files/subfolders should sync up to be at earliest the new specified date of the parent folder.